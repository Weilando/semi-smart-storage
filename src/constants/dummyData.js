const DummyItemList = [
  {id: 0, name: "Sparkling water"},
  {id: 1, name: "Apple juice"},
  {id: 2, name: "Craft beer"},
  {id: 3, name: "Milk"},
  {id: 5, name: "Coke"}
];

const DummyItemListShort = [
  {id: 0, name: "Sparkling water"},
  {id: 1, name: "Apple juice"},
  {id: 3, name: "Milk"}
];

const DummyStorageContent = [
  {id: 0, item: "Milk", unit: "1L", quantity: 2.7},
  {id: 1, item: "Coke", unit: "0.33L", quantity: 5},
  {id: 2, item: "Apple juice", unit: "0.5L", quantity: 2},
  {id: 4, item: "Craft Beer", unit: "0.33L", quantity: 6},
  {id: 5, item: "Sparkling water", unit: "0.75L", quantity: 21}
];

const DummyStorageContentShort = [
  {id: 0, item: "Milk", unit: "1L", quantity: 2.7},
  {id: 2, item: "Sparkling water", unit: "0.75L", quantity: 21}
];

const DummyStorageList = [
  {id: 0, name: 'Fridge'},
  {id: 1, name: 'Basement'},
  {id: 3, name: 'Garage'},
];

const DummyStorageListShort = [
  {id: 0, name: 'Basement'},
  {id: 3, name: 'Fridge'},
];

const DummyUnitList = [
  {id: 0, name: "0.33L"},
  {id: 1, name: "0.5L"},
  {id: 2, name: "0.75L"},
  {id: 3, name: "1.0L"}
];

const DummyUnitListShort = [
  {id: 0, name: "0.33L"},
  {id: 2, name: "0.75L"},
  {id: 3, name: "1.0L"}
];

export { DummyItemList, DummyStorageContent, DummyStorageList, DummyUnitList };
export { DummyItemListShort, DummyStorageContentShort, DummyStorageListShort, DummyUnitListShort };
