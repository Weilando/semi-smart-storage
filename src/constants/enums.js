// api-operations
const AddMode = {
  ITEM_LIST: 0, STORAGE_LIST: 1, STORAGE_CONTENT: 2, UNIT_LIST: 3
}

const DeleteMode = {
  ITEM_LIST: 0, STORAGE_LIST: 1, STORAGE_CONTENT: 2, UNIT_LIST: 3
}

const FetchMode = {
  ITEM_LIST: 0, STORAGE_LIST: 1, STORAGE_CONTENT: 2, UNIT_LIST: 3
}

const UpdateMode = {
  ITEM_LIST: 0,
  STORAGE_LIST: 1,
  STORAGE_CONTENT_DECREMENT: 2,
  STORAGE_CONTENT_INCREMENT: 3,
  STORAGE_CONTENT_EDIT: 4,
  UNIT_LIST: 5
}

// modal-modes
const ModalMode = {
  ITEM: "item",
  STORAGE: "storage",
  UNIT: "unit"
}

export { AddMode, DeleteMode, FetchMode, UpdateMode, ModalMode };
