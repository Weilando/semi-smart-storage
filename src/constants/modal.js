let ModalMode = {
  ITEM: "item",
  STORAGE: "storage",
  UNIT: "unit"
}

export { ModalMode }
