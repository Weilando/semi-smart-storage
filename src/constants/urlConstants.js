const BASEURL = "http://raspberrypi/api/";
const ADD = "add.php";
const DELETE = "delete.php";
const GET = "get.php";
const UPDATE = "update.php";

export { BASEURL };
export { ADD, DELETE, GET, UPDATE };
