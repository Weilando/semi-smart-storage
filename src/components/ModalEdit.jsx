import React from 'react';
import PropTypes from 'prop-types';
import { ModalMode } from '../constants/modal';
import '../styles/Modal.css';

function ModalEdit(props) {
  ModalEdit.propTypes = {
    show: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired,
    mode: PropTypes.oneOf([ModalMode.ITEM, ModalMode.UNIT, ModalMode.STORAGE]),
    buffer: PropTypes.string.isRequired,
    changeAction: PropTypes.func.isRequired,
    closeAction: PropTypes.func.isRequired,
    deleteAction: PropTypes.func.isRequired,
    updateAction: PropTypes.func.isRequired,
  }

  return(
    <div className={'modal '.concat(props.show ? 'show' : 'hide')}>
      <div className="modal-main">
        <span className="close" onClick={props.closeAction}>&times;</span>
        <h1>Settings for {props.mode} "{props.name}".</h1>


        <h2>Update name</h2>
        <form onSubmit={props.updateAction} className="Modal">
          <input
            type="text"
            className="Modal"
            value={props.buffer}
            onChange={props.changeAction}
          />
          <br/>
          <input
            type="submit"
            className="Modal"
            value="Update name"
            disabled={props.name === ''}
          />
        </form>

        <h2>Delete {props.mode}</h2>
        <p>
          {props.mode === ModalMode.STORAGE ? 'To delete the storage and all its content, click the button below. Items and units are not deleted.'
          : 'To delete the '.concat(props.mode, ', click the button below. All occurances in every storage will be deleted, too.')}
        </p>
        <button onClick={props.deleteAction} className="Modal">
          Delete {props.mode}
        </button>
      </div>
    </div>
  );
}

export default ModalEdit;
