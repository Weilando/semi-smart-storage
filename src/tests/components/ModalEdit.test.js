import React from 'react';
import renderer from 'react-test-renderer';
import { ModalMode } from '../../constants/enums';
import ModalEdit from '../../components/ModalEdit';

describe('snapshot tests for ModalEdit component', () => {
  it('should render visible ModalEdit component for storages correctly', () => {
    const changeAction = jest.fn();
    const closeAction = jest.fn();
    const deleteAction = jest.fn();
    const updateAction = jest.fn();


    const tree = renderer.create(
      <ModalEdit
        show={true}
        name="Fridge"
        mode={ModalMode.STORAGE}
        buffer="Fridgerator"
        closeAction={closeAction}
        changeAction={changeAction}
        deleteAction={updateAction}
        updateAction={updateAction}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })

  it('should render hidden ModalEdit component for storages correctly', () => {
    const changeAction = jest.fn();
    const closeAction = jest.fn();
    const deleteAction = jest.fn();
    const updateAction = jest.fn();


    const tree = renderer.create(
      <ModalEdit
        show={false}
        name="Fridge"
        mode={ModalMode.STORAGE}
        buffer="Fridge"
        closeAction={closeAction}
        changeAction={changeAction}
        deleteAction={updateAction}
        updateAction={updateAction}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })
});
