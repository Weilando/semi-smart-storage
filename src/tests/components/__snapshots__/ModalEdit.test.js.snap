// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for ModalEdit component should render hidden ModalEdit component for storages correctly 1`] = `
<div
  className="modal hide"
>
  <div
    className="modal-main"
  >
    <span
      className="close"
      onClick={[MockFunction]}
    >
      ×
    </span>
    <h1>
      Settings for 
      storage
       "
      Fridge
      ".
    </h1>
    <h2>
      Update name
    </h2>
    <form
      className="Modal"
      onSubmit={[MockFunction]}
    >
      <input
        className="Modal"
        onChange={[MockFunction]}
        type="text"
        value="Fridge"
      />
      <br />
      <input
        className="Modal"
        disabled={false}
        type="submit"
        value="Update name"
      />
    </form>
    <h2>
      Delete 
      storage
    </h2>
    <p>
      To delete the storage and all its content, click the button below. Items and units are not deleted.
    </p>
    <button
      className="Modal"
      onClick={[MockFunction]}
    >
      Delete 
      storage
    </button>
  </div>
</div>
`;

exports[`snapshot tests for ModalEdit component should render visible ModalEdit component for storages correctly 1`] = `
<div
  className="modal show"
>
  <div
    className="modal-main"
  >
    <span
      className="close"
      onClick={[MockFunction]}
    >
      ×
    </span>
    <h1>
      Settings for 
      storage
       "
      Fridge
      ".
    </h1>
    <h2>
      Update name
    </h2>
    <form
      className="Modal"
      onSubmit={[MockFunction]}
    >
      <input
        className="Modal"
        onChange={[MockFunction]}
        type="text"
        value="Fridgerator"
      />
      <br />
      <input
        className="Modal"
        disabled={false}
        type="submit"
        value="Update name"
      />
    </form>
    <h2>
      Delete 
      storage
    </h2>
    <p>
      To delete the storage and all its content, click the button below. Items and units are not deleted.
    </p>
    <button
      className="Modal"
      onClick={[MockFunction]}
    >
      Delete 
      storage
    </button>
  </div>
</div>
`;
